src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,441 @@
+    .jsonl and .csv files.
+
+    Args:
+        result (Evaluation):    ragas EvaluationResult to parse for scores
+                                for summary sheet and questions, references,
+                                and responses sheet.
+        excel_file (str):       Name of excel file to be created the summary
+                                and dataset sheets.
+    Returns:
+        None
+    """
+    summary_df = pd.DataFrame()
+    scores = [score["domain_specific_rubrics"] for score in result.scores]
+    summary_df["scores"] = scores
+
+    question_indices = [f"Q{i + 1}" for i in range(len(summary_df))]
+    question_indices.append("Average")
+    question_indices.append("Total Score")
+    question_indices.append("Median")
+
+    average = summary_df.mean(axis=0, numeric_only=True)
+    total_score = summary_df.sum(axis=0, numeric_only=True)
+    median = summary_df.median(axis=0, numeric_only=True)
+
+    summary_df.loc[len(summary_df)] = average
+    summary_df.loc[len(summary_df)] = total_score
+    summary_df.loc[len(summary_df)] = median
+
+    summary_df.insert(0, "question index", question_indices)
+
+    response_df = result.dataset.to_pandas()
+    response_df["scores"] = scores
+
+    with pd.ExcelWriter(excel_file) as writer:
+        # df with contents similar to those in the .jsonl and .csv output files
+        response_df.to_excel(writer, sheet_name="dataset", index=False)
+        summary_df.to_excel(writer, sheet_name="Summary", index=False)
+
+
+def write_results(
+    result: EvaluationResult, file_formats: List[str], output_dir: str, model_name: str
+) -> List[str]:
+    """
+    Writes results files for DK-Bench for each file format provided.
+    Files have the name {output_dir}/responses-{model_name}-{timestamp}.{file_type}"
+
+    Each of the entries in a file has the following fields:
+    {model_name, scores, evaluation_run, user_input, response, reference}
+
+    Args:
+        result (Evaluation):      ragas EvaluationResult to parse for scores
+                                  for summary sheet and questions, references,
+                                  and responses sheet.
+        file_formats (List[str]): List of file formats of files to write results to.
+        output_dir (str):         Directory for results to be written out to
+        model_name (str):         Model name to be used in the results file name.
+    Returns:
+        List[str]:                A list of strings that are the file names of the files
+                                  with responses and scores for DK-Bench
+    """
+    validate_output_file_formats(file_formats)
+    response_df = result.dataset.to_pandas()
+    response_df["model_name"] = model_name
+
+    scores = [score["domain_specific_rubrics"] for score in result.scores]
+    response_df["scores"] = scores
+
+    timestamp = datetime.now().isoformat()
+    response_df["evaluation_run"] = f"run-{timestamp}"
```

Comment:
Can we please get rid of this?

It seems like your goal here is to provide some kind of unique identifier, so maybe in the future we can track different runs.

Implementing something like this is out of scope for this PR. If we want to do this in the future, we would want to design it differently anyway. 

Storing this doesn't help us because each row gets the exact same field. At that point, the user can just infer which run it was based on the name of the file. So it would be more beneficial to just make sure that our naming conventions make sense.

Reply:
Sure I can just save the timestamp. I was trying to get ahead of having a unique identifier for each time the evaluation is called on a model on a dataset. This originated when we were iterating over multiple models and I wanted to be able to have each result have something unique for each model output.

That being said we can think about users consuming this data in future PRs and add on from there.

Reply:
@alimaredia I like the idea, and so we should separate it into another PR so it can get the love it deserves.